src/util.rs

```diff
@@ -125,4 +151,72 @@ macro_rules! deref {
   };
 }
 
+impl crate::ast::Net {
+  /// Transforms the net `x & ...` into `y & x ~ (arg y) & ...`
+  pub fn with_argument(&mut self, arg: crate::ast::Tree) {
+    use crate::ast::Tree;
+    let mut fresh = 0usize;
+    fn ensure_no_conflicts(tree: &Tree, fresh: &mut usize) {
+      match tree {
+        Tree::Ctr { lft, rgt, .. } | Tree::Op2 { lft, rgt, .. } | Tree::Mat { sel: lft, ret: rgt } => {
+          ensure_no_conflicts(lft, fresh);
+          ensure_no_conflicts(rgt, fresh);
+        }
+        Tree::Op1 { rgt, .. } => {
+          ensure_no_conflicts(rgt, fresh);
+        }
+        Tree::Var { nam } => {
+          if let Some(var_num) = var_to_num(nam) {
+            *fresh = (*fresh).max(var_num);
+          }
+        }
+        _ => (),
```

Comment:
nit: explicitly specify variants here, to make sure this is updated if other variants are added in the future